packages/cxl-ui/src/components/cxl-credential.js

```diff
@@ -0,0 +1,178 @@
+  }
+
+  @state({ type: Number }) correctionFactor;
+
+  @property({ type: Number, attribute: 'credential-id' }) credentialId = 0;
+
+  @property({ type: String, attribute: 'full-name' }) fullName = '';
+
+  /**
+   * @type {String}
+   * - 'course'
+   * - 'minidegree'
+   */
+  @property({ type: String }) title = '';
+
+  @property({ type: String }) type = '';
+
+  @property({ type: String }) date = '';
+
+  @property({ type: String, attribute: 'credential-url' }) credentialUrl = '';
+
+  @property({ type: Number }) scale = 0.5;
+
+  @property({ type: Number, attribute: 'max-width' }) maxWidth;
+
+  @property({ type: Number, attribute: 'parent-padding' }) parentPadding = 0;
+
+  @property({ type: Number, attribute: 'code-scale' }) codeScale = 8;
+
+  @query('#qr-code') qrCodeCanvas = null;
+
+  defaultUrl ='https://cxl.com/institute/credential'
+  
+  get url() {
+    return `${this.credentialUrl || this.defaultUrl}/${this.credentialId}`;
+  }
+
+  get formatedDate() {
+    const date = new Date(this.date);
+    const options = { year: 'numeric', month: 'long', day: 'numeric' };
+    return date.toLocaleDateString('en-US', options);
+  }
+
+  /**
+   *
+   * @param {number} w
+   * @param {number} h
+   * @returns {TemplateResult}
+   */
+  static _circle(stroke, d) {
+    return html`
+      <svg
+        width=${d}
+        height=${d}
+        viewBox="0 0 ${d + stroke} ${d + stroke}"
+        fill="none"
+        xmlns="http://www.w3.org/2000/svg"
+      >
+        <circle
+          cx="${(d + stroke) / 2}"
+          cy="${(d + stroke) / 2}"
+          r="${d / 2}"
+          fill="none"
+          stroke-width="${stroke}"
+          stroke="rgba(255,255,255,0.15)"
+        />
+      </svg>
+    `;
+  }
+
+  /**
+   * Generate background concentric circles decoration
+   *  @param {number} n
+   * @returns {TemplateResult}
+   */
+  static _circleBackground(n) {
+    // Initialize array of n circle diameters,
+    const arr = Array(n).fill(5.5);
+    return html` ${arr.map((stroke, i) => CXLCredentialElement._circle(stroke, stroke + i * 50))} `;
+  }
+
+  /**
+   * This function is called when this.scale is defined or changed. In order to
+   * prevent bad rendering on mobile devices, and because of how we use
+   * `transform: scale` to handle resizing, it calculates and applies a
+   * correction multiplier in case the viewport is smaller than the intended
+   * size of the certificate rendering.
+   *
+   * It works well, but is not meant to react to resizing a window or changing
+   * device orientation. But it will adjust, if the page is reloaded after.
+   */
+  _handleScaling() {
+    const width = this.maxWidth  || window.innerWidth - this.parentPadding;
+    const finalW = 1464 * this.scale;
+    this.correctionFactor = width < finalW || this.maxWidth ? width / finalW : 1;
+    this.style.height = `${1120 * this.scale * this.correctionFactor}px`;
+    this.style.width = `${1464 * this.scale * this.correctionFactor}px`;
```

Comment:
Can we avoid magic numbers or at least make them class properties and reuse here?

Reply:
Ok! Those are the fixed width and height dimensions I needed to use, from the design, to be able to keep everything in the right proportions. Ideally we would have them as public properties and be able  to make the entire thing scale properly, but that's something for another time. I'll add private class props for them.